adal/src/main/java/com/microsoft/aad/adal/DefaultTokenCacheStore.java

```diff
@@ -336,11 +336,14 @@ public void clearTokensForUser(String userId) {
 
         while (results.hasNext()) {
             final TokenCacheItem tokenCacheItem = results.next();
-                if (isAboutToExpire(tokenCacheItem.getExpiresOn())) {
+            if (tokenCacheItem.getExtendedExpiresOn() != null) {
+                if (isAboutToExpire(tokenCacheItem.getExtendedExpiresOn())) {
                     tokenItems.add(tokenCacheItem);
                 }
+            } else if (isAboutToExpire(tokenCacheItem.getExpiresOn())) {
+                tokenItems.add(tokenCacheItem);
```

Comment:
Can you send me the spec on this? I don't think this is the right behavior. Stale token should be only returned when I the outage mode is on and retry fails again with the three error code. Developer can call the public method to get all the tokens about to expires, it won't involve retry and you didn't check if the outage mode is on. Also server returns the extended expires on to indicate "we're capable of using the stale token", this will be a behavior change to all the new tokens we get from evo, they always return the extended expires on in JSON.